public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -165,6 +167,22 @@ const RulesFilter = ({ onFilterCleared = () => undefined }: RulesFilerProps) =>
             />
           </Field>
 
+          <Field
+            className={styles.dashboardPickerContainer}
+            label={<Label htmlFor="filters-dashboard-picker">Dashboard</Label>}
+          >
+            {/* The key prop is to clear the picker value */}
+            {/* DashboardPicker doesn't do that itself when value is undefined */}
+            <DashboardPicker
+              inputId="filters-dashboard-picker"
+              key={filterState.dashboardUid ? 'dashboard-not-defined' : 'dashboard-defined'}
```

Comment:
Is this correct? should not be the opposite?
` key={filterState.dashboardUid ?  'dashboard-defined' : 'dashboard-not-defined'}`

Reply:
Yes, it should 😅